### PR TITLE
[CWS] handles SECL rules using symlink paths

### PR DESCRIPTION
**Pull request by safchain**
### What does this PR do?

This PR introduce a way to handle symlink in SECL rules. It leverages operator override so that symlinks are resolved and target paths are used in SECL expressions. A callback in the mount resolver is used to resolve symlink in container.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] The appropriate `team/..` label has been applied, if known.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -98,6 +98,8 @@ require (
 	github.com/bhmj/jsonslice v0.0.0-20200323023432-92c3edaad8e2
 	github.com/blabber/go-freebsd-sysctl v0.0.0-20201130114544-503969f39d8f
 	github.com/cenkalti/backoff v2.2.1+incompatible
+	github.com/cespare/xxhash v1.1.0 // indirect
+	github.com/cespare/xxhash/v2 v2.1.2
```
Comment (vickenty) [inline go.mod:102]: We also have `murmur3` hash as a dependency already. Would it work instead of `xxhash`?

File: pkg/security/probe/applier.go [hunk 2]
```diff
@@ -97,7 +97,7 @@ func (rsa *RuleSetApplier) Apply(rs *rules.RuleSet, approvers map[eval.EventType
 			return nil, errors.Wrap(err, "failed to select probes")
 		}
 
-		if err := rsa.probe.FlushDiscarders(); err != nil {
+		if err := rsa.probe.OnRuleSetApplied(); err != nil {
```
Comment (vickenty) [inline pkg/security/probe/applier.go:100]: Should the error message below be updated too?

**Review by vickenty — APPROVED** [review]
Config changes LGTM, left a few questions inline.

File: pkg/ebpf/compiler/compiler.go [hunk 3]
```diff
@@ -12,7 +12,7 @@ package compiler
...
-#cgo LDFLAGS: -lz -ldl -lm -lrt -static-libstdc++
+#cgo LDFLAGS: -lz -ldl -lm -lrt -static-libstdc++ -ltinfo
```
Comment (brycekahle) [inline pkg/ebpf/compiler/compiler.go:15]: Can you explain why this linker flag is needed now, whereas it wasn't needed before?

Comment (safchain) [reply pkg/ebpf/compiler/compiler.go:15]: my bad, will remove it

File: pkg/security/probe/applier.go (hunk 2 above)
Comment (safchain) [reply pkg/security/probe/applier.go:100]: yes, I'll update, thx

File: go.mod (hunk 1 above)
Comment (safchain) [reply go.mod:102]: xxhash is reported to be faster than murmurhash in some article. That is being said, the path where I use it is maybe not that critical and maybe murmurhash will do the job. I'll test it

**Pull request closed without merging**